README.md

```diff
@@ -144,47 +199,95 @@ OptionSwitchItem(
 
 ```swift
-
 final class PulseTool: Tool {
     var name: String { "Pulse" }
 
-    func presentPreview(from viewController: UIViewController) {
-        let pulse = UIHostingController(rootView: ConsoleView())
-        let pulseNavigation = UINavigationController(rootViewController: pulse)
-        viewController.present(pulseNavigation, animated: true)
+    var content: any View {
+        ConsoleView()
     }
 }
 ```
 
+When creating custom `Tool` objects. You can use the `ToolTable` object which will use `SentinelListView` as the main View which is a ScrollView with a predefined number of items which can be shown. To create a `ToolTable` you need to provide a name for that tool, and the section items. Section items are items of type `ToolTableSection` which can have a section title, and an array of items of type `ToolTableItem`. `ToolTableItem` is an Enum with five different states which can be shown just by providing the required information for it:
+ - **navigation** - creates a View with title + arrow right which will lead to a different screen when tapped
+ - **toggle** - creates a View with a Toggle which will change the state of the provided property when tapped
+ - **customInfo** - creates a View with a title + value
+ - **performance** - creates a View with a title + value which will be updated each second
+ - **custom** - creates a View whichever you provide, gives the ability to show a custom View. The item will need to conform to the `CustomToolTableItem` protocol.
+
+```swift
+struct ColorChangeToolTableItem: CustomToolTableItem {
+
+    var title: String {
+        "Color change"
+    }
+
+    var content: any View {
+        ColorChangeToolView()
+    }
+}
+
+var colorChangeTool: Tool {
+    ToolTable(
+        name: "Color Change Tool",
+        sections: [
+            ToolTableSection(title: "Color change", items: [ToolTableItem.custom(ColorChangeToolTableItem())])
+        ]
+    )
+}
+```
+
 ### Available custom tools
 
+#### CustomInfoTool
+
 *CustomInfoTool* is used on the `Device`, and `Application` tabs. Its primary use is to list out properties and their values.
 
-*CustomLocationTool* is used to change the current user's location. After changing the location, the application will have to be restarted.
+#### TextEditingTool
 
 *TextEditingTool* is used to give the ability to edit a value. Where you have to provide a `getter`, and a `setter` for the property you want to change dynamically.
 
+#### UserDefaultsTool
+
 *UserDefaultsTool* is used to give an overview of all the `UserDefaults` properties, their values, and the ability to delete properties.
 
+#### iOS only
+
+#### CustomLocationTool
+
+*CustomLocationTool* is used to change the current user's location. After changing the location, the application will have to be restarted.
+
+#### EmailSenderTool
+
 *EmailSenderTool* is used to let the user send emails with attachments from the app with ease.
```

Comment:
As mentioned before, I would change a description of those and describe available tool for the developers (which are also available via Podspec).